core/src/main/java/org/infinispan/remoting/inboundhandler/TopologyMode.java

```diff
@@ -0,0 +1,70 @@
+package org.infinispan.remoting.inboundhandler;
+
+import org.infinispan.statetransfer.StateTransferLock;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * It checks or waits until the required topology is installed.
+ *
+ * @author Pedro Ruivo
+ * @since 8.0
+ */
+public enum TopologyMode {
```

Comment:
I don't really like the name, it suggests a regular "passive" enum that is used in a `switch` statement, like `CacheMode`, while this is just a holder for functionality like `ValueMatcher`.